native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe/serializer/json/JsonSerializer.cs

```diff
@@ -0,0 +1,178 @@
+            var payload = System.Text.Encoding.UTF8.GetString(data);
+            var jsonDataWithSchema = JsonDataWithSchema.Build(schemaDefinition, payload);
+            Validate(jsonDataWithSchema);
+        }
+
+        /// <inheritdoc />
+        public void Validate(object data)
+        {
+            var schema = GetSchema(data);
+            var dataNode = GetDataNode(data);
+            JsonValidator.ValidateDataWithSchema(schema, dataNode);
+        }
+
+        /// <inheritdoc />
+        public void SetAdditionalSchemaInfo(object data, ref GlueSchemaRegistrySchema schema)
+        {
+            // No-op
+            // Currently we do not need to modify schema object for Json message
+        }
+
+        private static bool IsWrapper(object data)
+        {
+            return data is JsonDataWithSchema;
+        }
+
+        private static byte[] WriteBytes(JsonNode data)
+        {
+            var options = new JsonWriterOptions();
+            using var stream = new MemoryStream();
+            using var writer = new Utf8JsonWriter(stream, options);
+
+            data.WriteTo(writer, new JsonSerializerOptions());
+            writer.Flush();
+            var array = stream.ToArray();
+            return array;
+        }
+
+        private static JsonSchema GetSchema(object data)
+        {
+            try
+            {
+                if (IsWrapper(data))
+                {
+                    var dataObject = (JsonDataWithSchema)data;
+                    var getSchemaTask = JsonSchema.FromJsonAsync(dataObject.Schema);
+                    return getSchemaTask.Result;
+                }
+                var schema = data is string dataJsonString
+                    ? JsonSchema.FromSampleJson(dataJsonString)
+                    : JsonSchema.FromType(data.GetType());
+                return schema;
+            }
+            catch (Exception e)
+            {
+                throw new AwsSchemaRegistryException(
+                    "Could not generate schema from the type provided " + data.GetType(), e);
+            }
+        }
+
+        private static JsonNode GetSchemaNode([NotNull] object data)
+        {
+            if (IsWrapper(data))
+            {
+                var dataObject = (JsonDataWithSchema)data;
+                return ConvertToJsonNode(dataObject.Schema);
+            }
+
+            try
+            {
+                var schema = data is string dataJsonString
+                    ? JsonSchema.FromSampleJson(dataJsonString)
+                    : JsonSchema.FromType(data.GetType());
+                var schemaNode = JsonNode.Parse(schema.ToJson());
+                return schemaNode;
+            }
+            catch (Exception e)
+            {
+                throw new AwsSchemaRegistryException(
+                    "Could not generate schema from the type provided " + data.GetType(), e);
+            }
+        }
+
+        private static JsonNode GetDataNode([NotNull] object data)
```

Comment:
Please test compatibility between Java and C#,

Java serialized data can be de-serialized by C# and vice-versa.

Reply:
I have verified it, the serialized bytes are the same.